src/passes/DeadArgumentElimination.cpp

```diff
@@ -57,6 +57,14 @@ struct DAEFunctionInfo {
   // Map of all calls that are dropped, to their drops' locations (so that
   // if we can optimize out the drop, we can replace the drop there).
   std::unordered_map<Call*, Expression**> droppedCalls;
+  // Whether this function contains any tail calls (including indirect tail
+  // calls) and the set of functions this function tail calls. Tail callers and
+  // tail callees cannot have their dropped returns removed because of the
+  // constraint that tail-callees must have the same return type as
+  // tail-callers. TODO: allow the removal of dropped returns from tail-callers
+  // if their tail-callees can have their returns removed as well.
```

Comment:
Are indirect tail calls fully handled by this? It's not clear to me why it's useful to track which functions are tail called if we can't know which are indirectly tail called. It's just a partial set.

I suspect that if the tail call feature is enabled, we may need to disable part of this pass?

Reply:
Functions in the table are not modified by this pass, so indirect tail calls should be ok just like indirect normal calls. We track which functions are tail called or perform tail calls because those functions need to be skipped during return value optimization on top of the functions that are already skipped because they can be indirectly called. Note that it is still safe to optimize the dead arguments of tail called and tail calling functions.